Remove metrics publishers

ReubenBond opened the pull request:
This partially addresses #3976

Snapshots of current system behavior are published using various interfaces. This removes that publishing mechanism.

sergeybykov commented:
@dotnet-bot test functional

xiazen commented on src/Orleans.Core/Statistics/ClientStatisticsManager.cs:
```diff
@@ -3,47 +3,36 @@
             IOptions<MonitoringStorageOptions> storageOptions,
             SerializationManager serializationManager, 
             IServiceProvider serviceProvider,
-            IHostEnvironmentStatistics hostEnvironmentStatistics,
```
why removing this two. Doesn't statistics still reports cpu, memory and all that ?

xiazen replied:
Oh I guess these two are here only for constructing the default metric publisher.
Can you make sure statistics still reports `IHostEnvironmentStatistics ` and `IAppEnvironmentStatistics ` ? since those are pretty important statistics

ReubenBond replied:
These values are only published to the metrics table, so we can remove them since they're no longer used.

xiazen commented on src/Orleans.Core/Statistics/ClientTableStatistics.cs:
```diff
@@ -1,142 +0,0 @@
-
-namespace Orleans.Runtime
-{
-    internal class ClientTableStatistics : IClientPerformanceMetrics, IDisposable
```
why is this removed? isn't this statistics, not metrics?

ReubenBond replied:
The class name is statistics, but the interface is metrics. This is only used for snapshot metrics

ReubenBond commented:
Fixed test: metrics & stats values were mixed up in the test grain.

ReubenBond commented:
Simplified some classes and refactored a little, including removing something which has bothered me for a long time:
`Silo.cs:`
```diff
-            siloStatistics.MetricsTable.ActivationDirectory = activationDirectory;
-            siloStatistics.MetricsTable.ActivationCollector = catalog.ActivationCollector;
-            siloStatistics.MetricsTable.MessageCenter = messageCenter;
```

xiazen commented on src/Orleans.Core/Statistics/IPerformanceMetrics.cs:
```diff
@@ -29,60 +28,13 @@ public interface ICorePerformanceMetrics
-        /// </summary>
-        long ClientCount { get; }
-
+    {
```
why do we remove IClientPerformanceMetrics and keep `ISiloPerformanceMetrics ` and `ICorePerformanceMetrics ` ? 

Especially `ICorePerformanceMetrics`, all of whose metrics are covered by `SiloRuntimeStatistics` ? 

Is it because other component is using it , such as Scheduler and WorkerPool? If yes, can they use `IAppEnvironmentStatistics` and `IHostEnvironmentStatistics` instead ?

ReubenBond replied:
My bad, I explained it here: https://github.com/dotnet/orleans/issues/3976#issuecomment-364274529

ReubenBond commented:
@dotnet-bot test functional

ReubenBond commented:
@xiazen I removed the class entirely now

sergeybykov commented:
@dotnet-bot test functional
